qa/rpc-tests/quality_voiding.py

```diff
@@ -0,0 +1,330 @@
+        mark_logs("Height : {}".format(res[1]['vout'][1]['maturityHeight']), self.nodes, DEBUG_MODE)
+        ud = self.nodes[1].getunconfirmedtxdata(bwt_address)
+        assert_equal(bwt_amount_2, ud['bwtImmatureOutput'])
+        mark_logs("Immature output : {}".format(ud['bwtImmatureOutput']), self.nodes, DEBUG_MODE)
+
+        mark_logs("Node 1 reconsider last invalidated block", self.nodes, DEBUG_MODE)
+        self.nodes[1].reconsiderblock(block_inv)
+        sync_mempools(self.nodes[1:NUMB_OF_NODES])
+        sync_blocks(self.nodes[1:NUMB_OF_NODES])
+
+        # Checking all certificate in mempool
+        mark_logs("Check certificates is not in mempool anymore", self.nodes, DEBUG_MODE)
+        assert_equal(False, cert_1_epoch_0 in self.nodes[1].getrawmempool())
+        assert_equal(False, cert_2_epoch_0 in self.nodes[1].getrawmempool())
+        assert_equal(False, cert_3_epoch_0 in self.nodes[1].getrawmempool())
+
+        height = self.nodes[0].getblockcount()
+        mark_logs("Check certificates in chain", self.nodes, DEBUG_MODE)
+        assert_true(cert_1_epoch_0 in self.nodes[1].getblock(self.nodes[1].getblockhash(height - 1), True)['cert'])
+        assert_true(cert_2_epoch_0 in self.nodes[1].getblock(self.nodes[1].getblockhash(height - 1), True)['cert'])
+        assert_true(cert_3_epoch_0 in self.nodes[1].getblock(self.nodes[1].getbestblockhash(), True)['cert'])
+
+        # Check balance
+        bwt_address = self.nodes[1].getrawcertificate(cert_3_epoch_0, 1)['vout'][1]['scriptPubKey']['addresses'][0]
+        res = self.nodes[1].listtxesbyaddress(bwt_address)
+        assert_equal(res[0]['scid'], scid)
+        mark_logs("Height : {}".format(res[1]['vout'][1]['maturityHeight']), self.nodes, DEBUG_MODE)
+        ud = self.nodes[1].getunconfirmedtxdata(bwt_address)
+        assert_equal(bwt_amount_3, ud['bwtImmatureOutput'])
+        mark_logs("Immature output : {}".format(ud['bwtImmatureOutput']), self.nodes, DEBUG_MODE)
+
+        utxos = self.nodes[1].listunspent()
+        assert_equal(len(utxos), 1)
+        assert_equal(utxos[0]['txid'], cert_3_epoch_0)
+
+        mark_logs("Reconnect network and check balance.", self.nodes, DEBUG_MODE)
+        stop_nodes(self.nodes)
+        wait_bitcoinds()
+        self.setup_network(False)
+
+        bwt_address = self.nodes[1].getrawcertificate(cert_3_epoch_0, 1)['vout'][1]['scriptPubKey']['addresses'][0]
+        res = self.nodes[1].listtxesbyaddress(bwt_address)
+        assert_equal(res[0]['scid'], scid)
+        mark_logs("Height : {}".format(res[1]['vout'][1]['maturityHeight']), self.nodes, DEBUG_MODE)
+        ud = self.nodes[1].getunconfirmedtxdata(bwt_address)
+        assert_equal(bwt_amount_3, ud['bwtImmatureOutput'])
+        mark_logs("Immature output : {}".format(ud['bwtImmatureOutput']), self.nodes, DEBUG_MODE)
+
+        utxos = self.nodes[1].listunspent()
+        assert_equal(len(utxos), 1)
+        assert_equal(utxos[0]['txid'], cert_3_epoch_0)
+
```

Comment:
We can check that the certificates with lower quality have been properly voided via the command:
    self.nodes[0].gettxout(cert_id, n)
where n is the output number: 0 for the change, never to be voided, and 1 for the bwt, which must be null for all certs but the top quality

Reply:
Done.